src/unix/solarish/compat.rs

```diff
@@ -169,3 +170,33 @@ pub unsafe fn forkpty(
 
     0
 }
+
+pub unsafe fn getpwent_r(
+    pwd: *mut passwd,
+    buf: *mut ::c_char,
+    buflen: ::size_t,
+    result: *mut *mut passwd,
+) -> ::c_int {
+    let res = native_getpwent_r(pwd, buf, buflen.try_into().unwrap());
```

Comment:
I haven't checked deeply but this `TryInto` may break our MSRV?

Reply:
It might be better just to clamp the value at `::c_int::MAX`.  It's not incorrect to do so, because we would just not be using the entire (apparently massive) buffer passed by the user.

Reply:
Yes, `TryInto` needs rust 1.34 or above. The current MSRV is 1.13.

@jclulow Would you like to test this on illumos? (I am sorry I haven't got a VM to work...) Normally this bufsize is obtained through `sysconf(_SC_GETPW/GR_R_SIZE_MAX)` at runtime. If the value is in the range of `int` on illumos, then we can use `as` to cast it here

Reply:
I think we only care about the type, not what values users might or might not pass, so probably something like this would work:

```rust
buflen.min(::c_int::MAX as ::size_t) as ::c_int
```

That way I think you would be able to pass any legal value up to and including the maximum `int` value.  The argument in the illumos libc version is definitely an `int`:

```
struct passwd *
getpwent_r(struct passwd *result, char *buffer, int buflen)
{
        nss_XbyY_args_t arg;
        char            *nam;
...
```

Reply:
And yes, either @pfmooney or myself can definitely test the final patch on an illumos system!